### ISSUE TITLE
Success message missing when reveal_type() used.

### ISSUE THREAD

**Issue by mrolle45** [python/mypy#12302]
**Bug Report**

I have something like `if TYPE_CHECKING: reveal_type(foo)` in a module (in 5 places).
mypy finds no errors, but the line that says "success" is missing.

Here's two sets of output from mypy, first with reveal_type() in the module, and second without.

```
D:\Solutions\trunk\Python\Sass>mypy .
solver\common.py:299: note: Revealed type is "Any"
solver\common.py:366: note: Revealed type is "Any"
solver\common.py:521: note: Revealed type is "solver.instruct.InstrTab"
solver\common.py:522: note: Revealed type is "Any"
solver\common.py:523: note: Revealed type is "Any"

D:\Solutions\trunk\Python\Sass>mypy .
Success: no issues found in 52 source files
```

What I think is correct output, in the first case, is
```
D:\Solutions\trunk\Python\Sass>mypy .
solver\common.py:299: note: Revealed type is "Any"
solver\common.py:366: note: Revealed type is "Any"
solver\common.py:521: note: Revealed type is "solver.instruct.InstrTab"
solver\common.py:522: note: Revealed type is "Any"
solver\common.py:523: note: Revealed type is "Any"
Success: no issues found in 52 source files
```

**Your Environment**

- Mypy version used: 0,92`
- Python version used: 3.7
- Operating system and version: Windows 10

**Comment by KotlinIsland** [comment]
playground: https://mypy-play.net/?mypy=latest&python=3.10&gist=ace38b94eb12ec3051736e89dc4037b5

**Comment by mrolle45** [comment]
I wrote a replacement for the code in main():
```
    code = 0
    if messages:
        n_errors, n_files = util.count_stats(messages)
        code = 1
    if blockers: code = 2
    if options.error_summary:
        if messages:
            if n_errors:
                summary = formatter.format_error(
                    n_errors, n_files, len(sources), blockers=blockers,
                    use_color=options.color_output
                )
                stdout.write(summary + '\n')
        if code == 0:
            stdout.write(formatter.format_success(len(sources), options.color_output) + '\n')
        stdout.flush()
```

You can use this code.

By the way, why are notes a cause for failure?

**Comment by KotlinIsland** [comment]
Here is a workaround:
```py
from typing import *

error: Literal[": error:"]
reveal_type(error)
```
```
> mypy main.py
main.py:4: note: Revealed type is "Literal[': error:']"
Found 1 error in 1 file (checked 1 source file)
```
https://mypy-play.net/?mypy=latest&python=3.10&gist=50bd436765ad217d1cc28568221c7c5f

**Comment by 97littleleaf11** [comment]
@KotlinIsland I think we can leave this corner case behind.

**Comment by KotlinIsland** [comment]
What I meant was, it is a workaround to the issue in the OP 😂
